File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/CompartmentDefinition.py
from typing import Optional, List
from pydantic import *
from ..base import *

class CompartmentDefinition_Resource(BackboneElement):
    code: str
    param: Optional[List[str]] = None
    documentation: Optional[str] = None

class CompartmentDefinition(DomainResource):
    description: Optional[str] = None
    date: Optional[str] = None
    publisher: Optional[str] = None
    purpose: Optional[str] = None
    name: str
    use_context: Optional[List[UsageContext]] = None
    experimental: Optional[bool] = None
    search: bool
    status: str
    resource: Optional[List[CompartmentDefinition_Resource]] = None
    url: str
    code: str
    version: Optional[str] = None
    contact: Optional[List[ContactDetail]] = None

File: test/aidbox_sdk/snapshots/dotnet/hl7-fhir-r4-core/CompartmentDefinition.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class CompartmentDefinition : DomainResource, IResource
{
    public string? Description { get; set; }
    public string? Date { get; set; }
    public string? Publisher { get; set; }
    public string? Purpose { get; set; }
    public required string Name { get; set; }
    public Base.UsageContext[]? UseContext { get; set; }
    public bool? Experimental { get; set; }
    public required bool Search { get; set; }
    public required string Status { get; set; }
    public CompartmentDefinition_Resource[]? Resource { get; set; }
    public required string Url { get; set; }
    public required string Code { get; set; }
    public string? Version { get; set; }
    public Base.ContactDetail[]? Contact { get; set; }

    public class CompartmentDefinition_Resource : BackboneElement
    {
        public required string Code { get; set; }
        public string[]? Param { get; set; }
        public string? Documentation { get; set; }
    }
}